File: RCE_HeroUI/src/rooms/models/Room.ts
import { RoomResponseDTO } from "../dtos/RoomResponseDTO";

export class Room {
  id: string;
  providerId: string;
  name: string;
  description?: string;
  theme?: string;
  minPlayers: number;
  maxPlayers: number;
  durationMinutes?: number;
  createdAt?: string;

  constructor(dto: RoomResponseDTO) {
    this.id = dto.id;
    this.providerId = dto.providerId;
    this.name = dto.name;
    this.description = dto.description;
    this.theme = dto.theme;
    this.minPlayers = dto.minPlayers;
    this.maxPlayers = dto.maxPlayers;
    this.durationMinutes = dto.durationMinutes;
    this.createdAt = dto.createdAt;
  }
}

File: RCE_Providers/Rooms/Entities/Room.cs
using RCE_Providers.EscapeRoomProviders.Entities;

namespace RCE_Providers.Rooms.Entities;

public class Room
{
    public Guid Id { get; set; }
    public Guid ProviderId { get; set; }
    public string Slug { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Theme { get; set; } = string.Empty;
    public required int MinPlayers { get; set; }
    public required int MaxPlayers { get; set; }
    public int DurationMinutes { get; set; } = 0;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public EscapeRoomProvider? Provider { get; set; }
}
